### ISSUE TITLE
Re-order off-topic channels in !ot embed

### ISSUE THREAD

**Issue by minalike** [python-discord/meta#157]
### Amendment type

Content Update

### Content type

Tag

### Please provide the location of the content

!ot

### What amendment do you think should be made to this content, and why?

`!ot` is very often used to re-direct users to off-topic channels when their discussions are out of scope for whatever channel they are in. Currently the channels in the embed are listed in numerical order: 0, 1, 2. 

Users then head over to `#ot0` and continue their conversation, often while there are existing ongoing conversations already in the channel. `#ot0` already gets a lot of traffic compared to `#ot1` and `#ot2`. The multiple overlapping conversations would be mitigated if members used the other 2 channels more.

My proposal was to have the below embed prioritize the off-topic channel with the oldest activity first, and most recent activity last. Other staff have suggested we may just reverse the current order to: 2, 1, 0.

Personally when directing others to off-topic I pick and mention whichever one has no recent activity, instead of using `!ot`

Screenshot of current embed:
![image](https://user-images.githubusercontent.com/75038675/140529170-411d3b3c-7cf4-476a-9adf-032dddf2d3f6.png)

**Comment by janine9vn** [comment]
So, I checked with the core devs and a dynamic re-ordering isn't possible with our tag system. All of our tags are static markdown files and to try to get any dynamic updating involved would require a lot of work.

So is the current suggestion to just manually re-order them to:
- ot2
- ot1
- ot0
?

**Comment by Numerlor** [comment]
> So, I checked with the core devs and a dynamic re-ordering isn't possible with our tag system. All of our tags are static markdown files and to try to get any dynamic updating involved would require a lot of work.
> 
> So is the current suggestion to just manually re-order them to:


If dynamic ordering would be preferred I don't think remaking it as an info command instead of a tag (for example like the rules command) would be difficult. The invocation fuzzy matching would be lost with that but I think the common cases could be covered with aliases

**Comment by minalike** [comment]
> So is the current suggestion to just manually re-order them to:
> 
> * ot2
> * ot1
> * ot0
>   ?

Yes, though it has the potential caveat of interrupting conversations in `#ot2` I think it's still better than leaving it ordered 0, 1, 2.


> If dynamic ordering would be preferred I don't think remaking it as an info command instead of a tag (for example like the rules command) would be difficult. The invocation fuzzy matching would be lost with that but I think the common cases could be covered with aliases

That would be cool!

**Comment by godlygeek** [comment]
I think just reordering to ot2, ot1, ot0 is a quick, easy experiment to run. If we do that and see usage of ot2 significantly increase, then clearly the ordering makes a difference, and we could talk about whether the effort of a dynamically ordered command would be worth it. If we do the static ot2, ot1, ot0 ordering and it doesn't make any noticeable difference to ot2's usage, then we shouldn't bother with carrying the idea any further.

**Comment by minalike** [comment]
While message volume per channel is one thing to observe, my main idea for this is to prevent the conversation interruption that happens in ot0 on the regular. I see it all the time, users in pygen are redirected to the ot channels via `!ot`, and ot0's conversation is abruptly interrupted. Original participants in ot0 are like `¯\_(ツ)_/¯` and continue on and you then have overlapping conversations, or then direct the users to another ot channel. I _feel_ like this _could_ be somewhat mitigated by either a dynamic or static re-ordering of the channels in the !ot embed

So my point is let's not hinge the decision on solely the effect it has on ot2's message volume.

**Comment by godlygeek** [comment]
I do understand what your goal is, but if we reorder the static tag to put ot2 first and ot0 last, and ot2 doesn't start getting used more, then we've learned that the order in the embed doesn't affect people's behavior significantly, and so dynamic reordering wouldn't prevent conversations from being interrupted.

**Comment by lemonsaurus** [comment]
I like this idea. I actually think it makes sense to start with just reversing the order to see what the effect is on ot2, this is very measurable through usage stats and unique users.

then if we see the traffic shifting more towards ot2, we could probably turn this into a dynamic command that puts them in random order. perhaps weighted in favor of putting ot1 and ot2 as the first channel, since there's probably also traffic hitting ot0 from being higher in the channel list.

**Comment by vcokltfre** [comment]
Why do we want to favour a specific channel anyway? It's not like (iirc) advanced discussion vs careers discussion vs community meta where each has a specific purpose and one ends up being visited more because it's higher, since all OT channels are... well... off topic? AFAIK there's no benefit to load balancing the OT channel activity, especially since at any one point it will still be whichever channel is highest that gets the most people talking in it

**Comment by lemonsaurus** [comment]
you start with ot2 on top so we can see if anything changes. if it does, we load balance evenly. why? to minimize the interruptions described in the OP.

**Comment by lemonsaurus** [comment]
we could even list them in order of least recent activity, maybe?

**Comment by dawnofmidnight** [comment]
Least recent activity sounds best to me, since the goal is to reduce interruptions, not necessarily balance out traffic.

**Comment by minalike** [comment]
@vcokltfre to clarify my proposal is for some method of re-ordering in the embed, not re-ordering the actual channels themselves

**Comment by minalike** [comment]
As decided on Nov 14th's staff meeting, python-discord/bot#1956 is a hard-coded re-ordering of the channels in the `!ot` embed. Joe pulled the following statistics for first off-topic channel used by members joined in the last 30 days.

**Comment by minalike** [comment]
As decided on Nov 14th's staff meeting, python-discord/bot#1956 is a hard-coded re-ordering of the channels in the `!ot` embed. Joe pulled the following statistics from Metricity for the first off-topic channel used by members joined in the last 30 days:

```sql
SELECT channel_id, COUNT(1) FROM (SELECT id, (SELECT channel_id FROM messages WHERE messages.author_id = users.id AND ARRAY[messages.channel_id]::text[] <@ ARRAY['291284109232308226', '463035241142026251', '463035268514185226'] ORDER BY messages.created_at LIMIT 1) FROM users WHERE users.in_guild AND users.joined_at >= NOW() - INTERVAL '30 days') users_and_first_ot_msg GROUP BY 1;
```

```
     channel_id     | count
--------------------+-------
 291284109232308226 |   209  |  first posted in ot0
 463035241142026251 |    37  |  first posted in ot1
 463035268514185226 |    28  |  first posted in ot2
                    | 19264  |  did not post in off-topic channels
```

We can run this query again in the future to measure whether the re-ordering of the channels has any effect, and decide whether or not to further action the idea of dynamic ordering based on oldest recent activity.

**Comment by jb3** [comment]
Stats from today, a bit over one month later after the change:
```
metricity=# SELECT channels.name, COUNT(1) AS num_messages FROM (SELECT id, (SELECT channel_id FROM messages WHERE messages.author_id = users.id AND ARRAY[messages.channel_id]::text[] <@ ARRAY['291284109232308226', '463035241142026251', '463035268514185226'] ORDER BY messages.created_at LIMIT 1) FROM users WHERE users.in_guild AND users.joined_at >= NOW() - INTERVAL '30 days') users_and_first_ot_msg LEFT JOIN channels ON channels.id = channel_id GROUP BY 1 ORDER BY num_messages DESC;
                name                | num_messages
------------------------------------+--------------
                                    |        18523
 ot0-the-fork-is-made-of-dough      |          160
 ot2-lizard-snake-guy               |           33
 ot1-human-equivalent-of-arch-linux |           32
(4 rows)
```

This shows that all in all this change has had minimal effect, and the majority of users continue to post in off topic 0 despite the reordering in the off topic embed.

I don't think this is any reason to revert the change, it's not *harming* off topic activity, but we could consider whether we want to go forward with the proposal of dynamically swapping the order based on most recent or just leave it as is.

Since we've seen minimal effect from the re-ordering, my personal view here is we should leave it as is, I think that complicating the command and storing last timestamps for an effect which has been negligible so far is a waste of work. I'll leave this for a week for other opinions, after that I'll close this as resolved.

**Comment by jb3** [comment]
For slightly easier digestible numbers, since implementing the change we've seen a 5% reduction (76% -> 71%) in the number of first off topic messages being in off-topic-0, and a 0.2% (1.4% -> 1.2%) reduction in overall new users sending a message in off topic. In my opinion, those changes are too minor to attribute to this change, and I don't think changing the order of the channels dynamically will in enhance the chances of people going to one of the quieter off topic channels first, the factors at play (already active channels, top channel in category) have a larger effect than ordering in the off topic embed.
